File: 05.01.26/count_words.py
# Count Words in a String in Python

def count_words(text):
    # Strip leading/trailing whitespace and split by whitespace
    words = text.strip().split()
    return len(words)

def count_words_detailed(text):
    words = text.strip().split()
    word_count = len(words)
    char_count = len(text)
    char_count_no_spaces = len(text.replace(" ", ""))
    
    return {
        'words': word_count,
        'characters': char_count,
        'characters_no_spaces': char_count_no_spaces
    }

def main():
    text = input("Enter a string: ")
    
    # Simple count
    word_count = count_words(text)
    print(f"\nNumber of words: {word_count}")
    
    # Detailed count
    stats = count_words_detailed(text)
    print(f"\nDetailed Statistics:")
    print(f"Words: {stats['words']}")
    print(f"Characters (with spaces): {stats['characters']}")
    print(f"Characters (without spaces): {stats['characters_no_spaces']}")

if __name__ == "__main__":
    main()

File: 05.01.26/CountWords.java
import java.util.*;

public class CountWords {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = sc.nextLine();
        
        String[] words = str.trim().split("\\s+");
        int wordCount = words.length;
        
        System.out.println("Number of words: " + wordCount);
        sc.close();
    }
}
